nft_ingester/src/bin/backfill/main.rs

```diff
@@ -0,0 +1,384 @@
+                        continue;
+                    }
+                };
+
+                if let Err(e) = consumer.consume_block(slot, raw_block.block).await {
+                    error!("Error processing slot {}: {}", slot, e);
+                }
+
+                // Increment slots_processed
+                let current_slots_processed = slots_processed.fetch_add(1, Ordering::Relaxed) + 1;
+
+                // Update progress bar position and message
+                progress_bar.inc(1);
+
+                let current_rate = {
+                    let rate_guard = rate.lock().unwrap();
+                    *rate_guard
+                };
+                progress_bar.set_message(format!(
+                    "Slots Processed: {} Current Slot: {} Rate: {:.2}/s",
+                    current_slots_processed, slot, current_rate
+                ));
+            }
+        });
+
+        worker_handles.push(handle);
+    }
+
+    // Spawn a task to update the rate periodically
+    let slots_processed_clone = slots_processed.clone();
+    let rate_clone = rate.clone();
+    let shutdown_token_clone = shutdown_token.clone();
+
+    tokio::spawn(async move {
+        let mut last_time = std::time::Instant::now();
+        let mut last_count = slots_processed_clone.load(Ordering::Relaxed);
+
+        loop {
+            tokio::time::sleep(std::time::Duration::from_secs(1)).await;
+
+            if shutdown_token_clone.is_cancelled() {
+                break;
+            }
+
+            let current_time = std::time::Instant::now();
+            let current_count = slots_processed_clone.load(Ordering::Relaxed);
+
+            let elapsed = current_time.duration_since(last_time).as_secs_f64();
+            let count = current_count - last_count;
+
+            let current_rate = if elapsed > 0.0 {
+                (count as f64) / elapsed
+            } else {
+                0.0
+            };
+
+            // Update rate
+            {
+                let mut rate_guard = rate_clone.lock().unwrap();
+                *rate_guard = current_rate;
+            }
+
+            // Update for next iteration
+            last_time = current_time;
+            last_count = current_count;
+        }
+    });
+
+    // Send slots to the channel
+    if !slots_to_process.is_empty() {
+        // Process only the specified slots
+        send_slots_to_workers(
+            slots_to_process,
+            source_db,
+            slot_sender.clone(),
+            shutdown_token.clone(),
+        )
+        .await;
+    } else {
+        // Process all slots from start_slot
+        send_all_slots_to_workers(
+            source_db,
+            slot_sender.clone(),
+            shutdown_token.clone(),
+            args.start_slot,
+        )
+        .await;
+    }
+
+    // Close the sender to signal that no more items will be sent
+    slot_sender.close();
+
+    // Wait for workers to finish
+    for handle in worker_handles {
```

Comment:
we have a lot of stuff like this, for me it seems reasonable to upgrade tokio for `join_all`. I hightly doubt in would break anything.

Reply:
also, we're awaiting for stuff but don't care about the result?